pkg/ingester/errors.go

```diff
@@ -482,7 +492,32 @@ func newIngesterErrSamplers(freq int64) ingesterErrSamplers {
 	}
 }
 
-func handlePushError(err error) error {
+func handlePushErrorWithGRPC(err error) error {
+	var (
+		ingesterErr ingesterError
+		errCode     = codes.Internal
+		wrappedErr  = err
+	)
+	if errors.As(err, &ingesterErr) {
+		switch ingesterErr.errorCause() {
+		case mimirpb.BAD_DATA:
+			errCode = codes.FailedPrecondition
+		case mimirpb.SERVICE_UNAVAILABLE:
+			errCode = codes.Unavailable
+		case mimirpb.INSTANCE_LIMIT:
+			errCode = codes.Unavailable
+			wrappedErr = log.DoNotLogError{Err: err}
+		case mimirpb.TSDB_UNAVAILABLE:
+			errCode = codes.Internal
+		}
+	}
+	return newErrorWithStatus(wrappedErr, errCode)
+}
+
+// TODO this method is needed only for the backwards compatibility.
+// It will be removed once httpgrpc has been completely removed
+// from both distributor and ingester.
+func handlePushErrorWithHTTPGRPC(err error) error {
```

Comment:
Nit, golang comments should start with method name.

```suggestion
// handlePushErrorWithHTTPGRPC is needed only for the backwards compatibility.
// TODO remove it once httpgrpc has been completely removed from both distributor and ingester.
func handlePushErrorWithHTTPGRPC(err error) error {
```

Ideally, the words `remove it once` should be `remove in 2.12.x` or similar, so it doesn't stay there forever.